### DISCUSSION TITLE
IRL Streaming - NGINX and RTMP Module?

### DISCUSSION THREAD

**Discussion by supercoopers** [SmoothWAN/SmoothWAN#65] [General & Help]
Hi!  Is there a way to use SmoothWAN with NGINX and RTMP?  
I'm using a raspberry pi 4 and an action cam that can stream video to an RTMP server.
I think I read something that SmoothWAN has NGINX, but I'm not sure how to setup or install RTMP on it!

**Comment by TalalMash** [comment]
Install with `wget` and `opkg` `--force-depends`: https://downloads.openwrt.org/releases/packages-19.07/aarch64_cortex-a72/packages/nginx-all-module_1.17.7-3_aarch64_cortex-a72.ipk

Create file: `/etc/init.d/ngix-rtmp`
```
#!/bin/sh /etc/rc.common
# Copyright (C) 2015 OpenWrt.org

START=80

USE_PROCD=1

G_OPTS="daemon off;"

NGINX_UTIL="/usr/bin/nginx-util"

eval $("${NGINX_UTIL}" get_env)

start_service() {

        procd_open_instance
        procd_set_param command /usr/sbin/nginx -c "/etc/nginx/rtmp.conf"
        procd_set_param stdout 1
        procd_set_param stderr 1
        procd_set_param respawn
        procd_close_instance
}
```
Create file `/etc/nginx/rtmp.conf` and customize to your needs: [DigitalOcean Guide]
(https://www.digitalocean.com/community/tutorials/how-to-set-up-a-video-streaming-server-using-nginx-rtmp-on-ubuntu-20-04)
You can also terminate Twitch, YoutTube and others here!:
```
rtmp {
  server {
    listen 1935;
    chunk_size 4096;
    application live {
      live on;
      record off;
    }
  }
}
```
Enable `nginx-rtmp`
```
service nginx-rtmp enable
service nginx-rtmp start
```
